src/lib/scheduler/src/sync/simple_latch.rs

```diff
@@ -1,7 +1,7 @@
 use std::sync::atomic::{AtomicU32, Ordering};
 use std::sync::Arc;
 
-use linux_api::errno::Errno;
+use nix::errno::Errno;
```

Comment:
Isn't this going backwards? Ie, we have been wanting to remove dependency on nix. Though I know [our guidance](https://shadow.github.io/docs/guide/coding.html#platform-libc-and-linux-crates) is not strict. Not sure but would `rustix` make more sense than `nix` or `linux-api`?

Reply:
I don't think we necessarily want to remove shadow's dependency on nix, it's just that we shouldn't be using libc types (which nix wraps) in syscall handlers (since libc and syscall types are often different) or in the shim (because of re-entrant functions). But it's still okay to use in other parts of shadow.

> For code that runs in the shadow process, where we are acting as a "normal" program that wants to interact with the kernel, generally prefer the highest-level interface that provides the necessary functionality. e.g. when creating worker threads in Rust, we generally use std::thread; there's no reason to use one of the lower level crates. Occasionally we need some functionality not provided in std though, in which case it makes sense to drop down to one of the lower level crates.

Here the rust standard library doesn't really have a nice way to work with errno values, so the next highest-level type is the nix `Errno` type.

In this case we need to use libc for futexes anyway (to support miri), and I think it makes sense to have a libc-based errno implementation (the one in nix) since we're making the syscall through libc. We also already use nix for other parts of the scheduler (such as setting CPU affinity). Switching the errno type to nix from libc-api also lets us remove linux-api as a dependency of the scheduler since we don't use it for anything else.

Reply:
OK, makes sense! Thanks for the explanations!